cmd/thanos/compact.go

```diff
@@ -113,9 +121,13 @@ func runCompact(
 			return errors.Wrap(err, "create compactor")
 		}
 
+		mutex := &sync.Mutex{}
 		ctx, cancel := context.WithCancel(context.Background())
 
 		f := func() error {
```

Comment:
Ok at this point I feel like we have to move those to separate function for readability. `runCompact` is too huge now.

Reply:
and name this more verbosly than just `f`